### PR TITLE
Styleguide

### PR DESCRIPTION
**Pull request by thecoolmichael**
style guide pull request

### CODE REVIEW

**Comment by Scholars-Mate** [comment]
We are probably going to want to squash this down to one commit before merging with master.

File: StyleGuide.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+* **External Documentation**
+* **Class Documentation**
+* **Interal Documentation & Naming**
+* **Indentation**
+* **Miscellaneous**
+---
+#### External Documentation:
+* At the top of each file created, the name of the team member working on this file and a brief description of the purpose of the file should be added.
+#### Class Documentation:
+* Each support class is to be preceded with a block comment that includes:
```
Comment (Scholars-Mate) [inline StyleGuide.md]: We should include an example of what the block comment looks like.  
We should also specify what documentation style we are using, e.g. Doxygen.

File: StyleGuide.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+   * Brief note on the classes purpose
+
+#### Internal Documentation & Naming:
+* It is encouraged to name variables and methods descriptively, but if variable names become scarce inline comments are necesary for future readability.
+* *Complex* pieces of code are encouraged to be commented heavily.
+
+
+#### Indentation:
+
+* RetroConnect is a **Tab** operated team with **tab stop = 4**. The use of spaces by a RetroConnect employee will result in immediate termination.
```
Comment (Scholars-Mate) [inline StyleGuide.md]: Why not 8 space tabs? Also related, how many columns should our lines be?

File: StyleGuide.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+#### Miscellaneous:
+* The following guidelanes are reccomended to follow when writing any code:
+  * Write programs with modularity; that is, create classes when appropriate, write methods that accomplish limited and well-defined tasks
+  * **Limited** or very monitored use of Global Variables
+  * Indent bodies of methods, loops and IF statements with consistent style
```
Comment (Scholars-Mate) [inline StyleGuide.md]: What style are we using for loops and if statements?

```cpp
if (condition) {
        // Statements
}

// or

if (condition)
{
        // Statements
}
```
Personally, I am partial to the first style.

File: StyleGuide.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# RetroConnect Team Style Guide ![Logo] ()
+# RetroConnect Team Style Guide ![Logo] (https://github.com/retroconnect/retroconnect/blob/Styleguide/images/RetroConnect.png)
```
Comment (Scholars-Mate) [inline StyleGuide.md]: We should probably just use a relative path, e.g. `/images/RetroConnect.png`

File: StyleGuide.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 # RetroConnect Team Style Guide 
-![Logo](https://github.com/retroconnect/retroconnect/blob/Styleguide/images/RetroConnect.png "RetroConnect")
+![Logo](https://github.com/retroconnect/retroconnect/blob/Styleguide/images/RetroConnect.png =100x20)
```
Comment (Scholars-Mate) [inline StyleGuide.md]: Maybe we should just resize the image in the repo before uploading.

File: StyleGuide.md (hunk 3 above)
Comment (seanconners) [reply StyleGuide.md]: First style definitely. Think of how many enter key presses could be saved!

**Pull request closed without merging**